client_test.go

```diff
@@ -135,6 +136,12 @@ func TestHandleErrorResp(t *testing.T) {
 				t.Errorf("Unexpected error: %v , expected: %s", err, tc.expected)
 				t.Fail()
 			}
+
+			e := &APIError{}
+			if !errors.As(err, &e) {
```

Comment:
according to the docs. It is supposed to be a double pointer [https://go.dev/blog/go1.13-errors](https://go.dev/blog/go1.13-errors)

> Note: although it may feel odd to take a pointer to a pointer, in this case it is correct. Think of it instead as taking a pointer to a value of the error type; it so happens in this case that the returned error is a pointer type.